packages/forge/src/governance/extensions/GovernorSorting.sol

```diff
@@ -0,0 +1,229 @@
+    }
+
+    function _sortItem(uint256 pos, int256[] memory netProposalVotes, uint256[] memory proposalIds)
+        internal
+        pure
+        returns (bool)
+    {
+        uint256 wMin = pos;
+        for (uint256 i = pos; i < netProposalVotes.length; i++) {
+            if (netProposalVotes[i] < netProposalVotes[wMin]) {
+                wMin = i;
+            }
+        }
+        if (wMin == pos) return false;
+        int256 votesTmp = netProposalVotes[pos];
+        netProposalVotes[pos] = netProposalVotes[wMin];
+        netProposalVotes[wMin] = votesTmp;
+        uint256 proposalIdsTmp = proposalIds[pos];
+        proposalIds[pos] = proposalIds[wMin];
+        proposalIds[wMin] = proposalIdsTmp;
+        return true;
+    }
+
+    /**
+     * @dev Accessor to sorted list of proposalIds in ascending order.
+     */
+    function sortedProposals(bool excludeDeletedProposals)
+        public
+        view
+        virtual
+        returns (uint256[] memory sortedProposalIdsReturn)
+    {
+        (uint256[] memory proposalIdList, VoteCounts[] memory proposalVoteCountsArray) =
+            excludeDeletedProposals ? allProposalTotalVotesWithoutDeleted() : allProposalTotalVotes();
+        int256[] memory netProposalVotes = new int256[](proposalIdList.length);
+        for (uint256 i = 0; i < proposalVoteCountsArray.length; i++) {
+            netProposalVotes[i] =
+                int256(proposalVoteCountsArray[i].forVotes) - int256(proposalVoteCountsArray[i].againstVotes);
+        }
+        for (uint256 i = 0; i < proposalIdList.length - 1; i++) {
+            // Only goes to length minus 1 because sorting the last item would be redundant
+            _sortItem(i, netProposalVotes, proposalIdList);
+        }
+        return proposalIdList;
+    }
+
+    /**
+     * @dev Setter for _sortedProposalIds, tiedAdjustedRankingPosition, _isTied, _lowestRanking,
+     * and _highestTiedRanking. Will only be called once and only needs to be called once because once the contest
+     * is complete these values don't change. Determines if a ranking is tied and also where the last
+     * iteration of a ranking is in the _sortedProposalIds list taking ties into account.
+     */
+    function setSortedAndTiedProposals() public virtual {
+        require(
+            state() == IGovernor.ContestState.Completed,
+            "GovernorSorting: contest must be to calculate sorted and tied proposals"
+        );
+        require(
+            setSortedAndTiedProposalsHasBeenRun == false,
+            "GovernorSorting: setSortedAndTiedProposals() has already been run and its respective values set"
+        );
+
+        _sortedProposalIds = sortedProposals(true);
+
+        int256 lastTotalVotes;
+        uint256 rankingBeingChecked = 1;
+        _highestTiedRanking = _sortedProposalIds.length + 1; // set as default so that it isn't 0 if no ties are found
+        for (uint256 i = 0; i < _sortedProposalIds.length; i++) {
```

Comment:
I know I already gave this advice and you haven't refactored this yet, but just want to note that since this is a storage array reading it's length in every iteration is even costlier than in the other example. Just writing this so it surely doesn't get missed ^^

Reply:
Yes, thank you this is really helpful!